tests/sources/test_slack.py

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+from unittest.mock import ANY, Mock, patch
+
+import pytest
+
+from connectors.logger import logger
+from connectors.sources.slack import SlackClient, SlackDataSource
+from tests.commons import AsyncIterator
+from tests.sources.support import create_source
+
+configuration = {
+    "token": "fake_token",
+    "auto_join_channels": False,
+    "fetch_last_n_days": 180,
+    "sync_users": True,
+}
+
+
+@pytest.fixture
+def slack_client():
+    client = SlackClient(configuration)
+    client.set_logger(logger)
+    yield client
+    client.close()
+
+
+@pytest.fixture
+def slack_data_source():
+    source = create_source(SlackDataSource, **configuration)
+    source.set_logger(logger)
+    yield source
+    source.close()
```

Comment:
You'll need to mark them both with `pytest_asyncio.fixture`, otherwise you'll get tons of warnings. We don't enforce tests to not break when warnings are emitted, but it'll happen really soon